engine/utils/event_processor.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <eventpp/eventqueue.h>
+#include <thread>
+#include "common/download_event.h"
+#include "common/event.h"
+
+namespace cortex::event {
+
+using Event = cortex::event::Event;
+using DownloadEvent = cortex::event::DownloadEvent;
+using EventQueue = eventpp::EventQueue<std::string, void(DownloadEvent)>;
+
+class EventProcessor {
+ public:
+  EventProcessor(std::shared_ptr<EventQueue> queue)
+      : event_queue_(std::move(queue)), running_(true) {
+    thread_ = std::thread([this]() {
+      while (running_) {
```

Comment:
Do we have a way to use `std::condition_variable` in this case?

Reply:
Update remove sleep